flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedSequenceNumberingViewReader.java

```diff
@@ -94,13 +99,27 @@ public void requestSubpartitionView(
 	}
 
 	@Override
-	public void addCredit(int creditDeltas) {
+	public boolean addCredit(int creditDeltas) {
 		numCreditsAvailable += creditDeltas;
+		return shouldAnnounceBacklog();
```

Comment:
why we need to trigger announce backlog while adding credit?
I assume since the added `creditDeltas` is always more than zero, then we have the chance to announce the backlog later via sending `BufferResponse`.

Reply:
You are right ```shouldAnnounceBacklog``` always return false, I will replace the function call with ```return false```. The reason why we need to change return type of addCredit is that we can't identify whether we are calling ```addCredit``` or ```resumeConsumption``` in ```PartitionRequestQueue```.